index.go

```diff
@@ -39,10 +41,48 @@ const (
 
 const compactionPageBytes = minSectorSize * 64
 
+type indexWriterSeries struct {
+	labels labels.Labels
+	chunks []*ChunkMeta // series file offset of chunks
+	offset uint32       // index file offset of series reference
+}
+
+type indexWriterSeriesSlice []*indexWriterSeries
+
+func (s indexWriterSeriesSlice) Len() int      { return len(s) }
```

Comment:
`sort.Slice` has a notable overhead. As we do this a lot I figured it's worth giving a native implementation.